src/core/comm/coll.h

```diff
@@ -97,6 +92,150 @@ struct Coll_Comm {
 
 typedef Coll_Comm* CollComm;
 
+class BackendNetwork {
+ public:
+  BackendNetwork();
```

Comment:
Do we need to declare a virtual destructor also? I know that would normally be necessary to get proper destruction of subclasses referenced through the static variable `BackendNetwork* backend_network`, but possibly virtual destructors are the default for abstract classes (?)

Reply:
You are correct, we need a virtual destructor, otherwise, the destructor of derived classes will not be called.